lib/marloss/store.rb

```diff
@@ -2,14 +2,16 @@
 
 module Marloss
   class Store # rubocop:disable Metrics/ClassLength
-    attr_reader :client, :table, :hash_key, :expires_key, :ttl
+    attr_reader :client, :table, :hash_key, :expires_key, :ttl, :unique_process_id
 
-    def initialize(table, hash_key, expires_key: "Expires", ttl: 30, client_options: {})
+    def initialize(table, hash_key, **options)
+      client_options = options[:client_options] || {}
       @client = Aws::DynamoDB::Client.new(client_options)
       @table = table
       @hash_key = hash_key
-      @expires_key = expires_key
-      @ttl = ttl
+      @expires_key = options[:expires_key] || "Expires"
+      @ttl = options[:ttl] || 30
+      @unique_process_id = options[:unique_process_id] || nil
```

Comment:
i would argue you could pass a non unique value here. lets change it to `custom_process_id`
```
@custom_process_id = options[:custom_process_id]
```

Reply:
actually now that i think about it we could do this:
```
attr_reader :client, ......., :process_id

def initialize(table, hash_key, **options)
  ...
  @process_id = options[:custom_process_id] || host_process_id
end

private def host_process_id
  hostname = `hostname`.chomp
  pid = Process.pid

  "#{hostname}:#{pid}"
end
```

change `process_id` to `host_ process_id` and set the custom || host at initialization

Reply:
Thanks for the quick response! I like the idea but that complicates the tests since the object is created before the stubs for `hostname` and `pid`.  Any preference on how to handle those?

Reply:
you would need to set the `custom_process_id` to nil in the main block and then set it in a separate context.

```
  let(:custom_process_id) { nil }
  let(:options) do
    { ttl: ttl, client_options: client_options, custom_process_id: custom_process_id }
  end
  let(:store) { described_class.new(table, hash_key, options) }

....

  context "with custom process id" do
    let(:custom_process_id) { "58a0b601" }

    it "should create the lock" do
      expect(ddb_client).to receive(:put_item).with(
        table_name: table,
        item: {
          hash_key => name,
          "ProcessID" => custom_process_id,
          "Expires" => expires
        },
        expression_attribute_names: {
          "#E" => "Expires",
          "#P" => "ProcessID"
        },
        expression_attribute_values: {
          ":now" => Time.now.to_i,
          ":process_id" => custom_process_id
        },
        condition_expression: "attribute_not_exists(#{hash_key}) OR #E < :now OR #P = :process_id"
      )

      store.create_lock(name)
    end
  end

....
```

^^ this spec should do the job 😄

Reply:
Ah yeah that's cleaner, updated and changed that stub to a `allow_any_instance_of`